src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVCustomReadFilter.java

```diff
@@ -0,0 +1,120 @@
+    private boolean nonSecondaryAlignmentsOnly;
+
+    // If set to true, filter out supplementary alignments reads.
+    private boolean nonSupplementaryAlignmentsOnly;
+
+    // If set to true, collect from reads that are first end of a pair.
+    private int whichEnd;
+
+    // If set non-zero value, only include reads passing certain mapping quality threshold.
+    private double MQPassingThreshold = 0.0;
+
+    // If set to non-zero value, only include reads with mate passing certain mapping quality threshold.
+    private double mateMQPassingThreshold = 0.0;
+
+    private final primaryFilter pfilter = new primaryFilter();
+    private final customFilter sfilter = new customFilter();
+
+
+    public SVCustomReadFilter(final boolean[] sixFilterOptions,
+                              final int whichEnd,
+                              final double mappingQualityThreshold,
+                              final double mateMappingQualityThreshold){
+        properlyPairedOnly = sixFilterOptions[0];
+        mappedReadsOnly = sixFilterOptions[1];
+        mateMappedReadsOnly = sixFilterOptions[2];
+        nonDupReadsOnly = sixFilterOptions[3];
+        nonSecondaryAlignmentsOnly = sixFilterOptions[4];
+        nonSupplementaryAlignmentsOnly = sixFilterOptions[5];
+
+        this.whichEnd = whichEnd;
+
+        MQPassingThreshold = mappingQualityThreshold;
+        mateMQPassingThreshold = mateMappingQualityThreshold;
+    }
+    @Override
+    public boolean test(final GATKRead read){
+        return pfilter.and(sfilter).test(read);
+    }
+
+    // similar to WellformedReadFilter except doesn't check for header but checks if mapping quality is available
+    private final class primaryFilter implements ReadFilter{
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public boolean test(final GATKRead read){
+            final ReadFilter primary = ReadFilterLibrary.PASSES_VENDOR_QUALITY_CHECK
+                                        .and(ReadFilterLibrary.VALID_ALIGNMENT_START)
+                                        .and(ReadFilterLibrary.VALID_ALIGNMENT_END)
+                                        .and(ReadFilterLibrary.HAS_READ_GROUP)
+                                        .and(ReadFilterLibrary.HAS_MATCHING_BASES_AND_QUALS)
+                                        .and(ReadFilterLibrary.READLENGTH_EQUALS_CIGARLENGTH)
+                                        .and(ReadFilterLibrary.SEQ_IS_STORED)
+                                        .and(ReadFilterLibrary.CIGAR_IS_SUPPORTED)
+                                        .and(ReadFilterLibrary.MAPPING_QUALITY_AVAILABLE);
+            return primary.test(read);
+        }
+    }
+
+    private final class customFilter implements ReadFilter{
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public boolean test(final GATKRead read){
+
+            boolean result = true;
```

Comment:
in this code, the read will be tested for all the conditions even if the first one fails. Change it so that the test returns `false` as quickly as possible in that case.